pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1822,21 +1730,23 @@ func (s *AWSCloud) updateInstanceSecurityGroupsForLoadBalancer(lb *elb.LoadBalan
 
 		permissions := []*ec2.IpPermission{permission}
 
-		if add {
-			changed, err := s.ensureSecurityGroupIngress(instanceSecurityGroupId, permissions)
+		if instanceSecurityGroupId != nil && groupId == instanceSecurityGroupId {
```

Comment:
Don't you need to dereference the pointers here?  Also I'm wondering if instanceSecurityGroupId should just be a string (vs a *string) - is it ever nil?